score/etl/sources/census/etl.py

```diff
@@ -0,0 +1,115 @@
+    state_fips_codes = get_state_fips_codes(data_path)
+    for fips in state_fips_codes:
+        # check if file exists
+        shp_file_path = data_path.joinpath(
+            "census", "shp", fips, f"tl_2010_{fips}_bg10.shp"
+        )
+        if not os.path.isfile(shp_file_path):
+            logging.info(f"Downloading {fips}")
+
+            # 2020 tiger data is here: https://www2.census.gov/geo/tiger/TIGER2020/BG/
+            # But using 2010 for now
+            cbg_state_url = f"https://www2.census.gov/geo/tiger/TIGER2010/BG/2010/tl_2010_{fips}_bg10.zip"
+            unzip_file_from_url(
+                cbg_state_url, data_path / "tmp", data_path / "census" / "shp" / fips
+            )
+
+        geojson_dir_path = data_path.joinpath(
+            "census",
+            "geojson",
+        )
+
+        if not os.path.isfile(geojson_dir_path.joinpath(fips + ".json")):
+            # ogr2ogr
+            logging.info(f"Encoding GeoJSON for {fips}")
+
+            # if it's running inside Docker container
+            if os.environ.get("DOCKER_CONTAINER"):
+                # TODO
+                pass
+            else:
+                # this is being run on virtualenv
+                # PWD is different for Windows
+                if os.name == "nt":
+                    pwd = "%cd%"
+                else:
+                    pwd = "${PWD}"
+                cmd = (
+                    'docker run --rm -it -v "'
+                    + pwd
+                    + '"/:/home osgeo/gdal:alpine-ultrasmall-latest ogr2ogr -f GeoJSON /home/data/census/geojson/'
+                    + fips
+                    + ".json /home/data/census/shp/"
+                    + fips
+                    + "/tl_2010_"
+                    + fips
+                    + "_bg10.shp"
+                )
+                os.system(cmd)
+
+        # generate CBG CSV table for pandas
+        ## load in memory
+        cbg_national_list = []  # in-memory global list
+        cbg_per_state_list: dict = {}  # in-memory dict per state
+        for file in os.listdir(geojson_dir_path):
+            if file.endswith(".json"):
+                logging.info(f"Ingesting geoid10 for file {file}")
+                with open(geojson_dir_path.joinpath(file)) as f:
+                    geojson = json.load(f)
+                    for feature in geojson["features"]:
+                        geoid10 = feature["properties"]["GEOID10"]
+                        cbg_national_list.append(str(geoid10))
+                        geoid10_state_id = geoid10[:2]
+                        if not cbg_per_state_list.get(geoid10_state_id):
+                            cbg_per_state_list[geoid10_state_id] = []
+                        cbg_per_state_list[geoid10_state_id].append(geoid10)
+
+        csv_dir_path = data_path / "census" / "csv"
+        ## write to individual state csv
+        for state_id in cbg_per_state_list:
```

Comment:
I'm having a little trouble following - what is being written here? Is the only data in this file going to be the CBG GEOID? 

(Either way, do you mind adding some more comments explaining it? Thanks!)